File: HmacCalculation/Hmac256Calculation.php
<?php
function hmac256Calculation($keyHmac, $data)
{
    return hash_hmac("sha256" , $data, $keyHmac, true);
}

function base64Encoding($input)
{
    return base64_encode($input);
}

echo 'HMAC 256 calculation' . PHP_EOL;

$hmac256KeyString = 'hmac256ForAesEncryption';
$plaintextString = 'The quick brown fox jumps over the lazy dog';
echo 'hmac256Key: ' . $hmac256KeyString . PHP_EOL;
echo 'plaintext:  ' . $plaintextString . PHP_EOL;

// calculate hmac
$hmac256 = hmac256Calculation($hmac256KeyString, $plaintextString);
echo 'hmac256 length: ' . strlen($hmac256) . ' (Base64) data: ' . base64Encoding($hmac256) . PHP_EOL;
?>

File: HmacCalculation/Hmac256Calculation.go
package main

import (
	"crypto/sha256"
  "crypto/hmac"
  "encoding/base64"
	"fmt"
  "strconv"
)

func main() {
  fmt.Printf("HMAC 256 calculation\n")

  hmac256KeyString := string("hmac256ForAesEncryption")
  plaintextString := string("The quick brown fox jumps over the lazy dog")
  fmt.Printf("hmac256Key: " + hmac256KeyString + "\n")
  fmt.Printf("plaintext: " + plaintextString + "\n")
   // get binary data
  plaintext := []byte(plaintextString)
  hmacKey := []byte(hmac256KeyString)

  // calculate hmac
  hmac256 := Hmac256Calculation(hmacKey, plaintext)
  hmacLength := len(hmac256)
  fmt.Printf("hmac256 length: " + strconv.Itoa(hmacLength) + " (Base64) data: " + Base64Encoding(hmac256) + "\n")
}

func Hmac256Calculation(hmacKey []byte, plaintext []byte)([]byte) {
    h := hmac.New(sha256.New, hmacKey)
    h.Write(plaintext)
    return h.Sum(nil)
}

func Base64Encoding(input []byte)(string) {
  return base64.StdEncoding.EncodeToString(input)
}
